### PR TITLE
[DOC] Mention behaviour of np.squeeze with one element

### LINKED ISSUE #15313: Squeeze array with one element returns an array with no shape
When squeezing an array that has only one element we get a non-dimensional array (instead of an array with one dimension).

### Reproducing code example:

```python
import numpy as np
arr = np.array([[1238]]).squeeze()
arr.shape
arr[0]
```

### Error message:
IndexError: too many indices for array

### Numpy/Python version information:
1.16.4 3.6.8 |Anaconda custom (64-bit)| (default, Feb 21 2019, 18:30:04) [MSC v.1916 64 bit (AMD64)]

### PR DESCRIPTION
**Pull request by bharatr21**
Fixes #15313 by mentioning the behaviour of `np.squeeze()` with only one / single element and adds one example

### CODE REVIEW

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -1472,6 +1474,17 @@ def squeeze(a, axis=None):
...
+    >>> x = np.array([[1234]])
+    >>> x.shape
+    (1, 1)
+    >>> np.squeeze(x)
+    array(1234)
+    >>> np.squeeze(x).shape
+    ()
+    >>> np.squeeze(x)[0]
+    Traceback (most recent call last):
+    ...
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py]: I don't think these last two examples are useful. Perhaps add `np.squeeze(x)[()]` to replace them?

File: numpy/core/fromnumeric.py [hunk 2]
```diff
@@ -1472,6 +1474,17 @@ def squeeze(a, axis=None):
     ValueError: cannot select an axis to squeeze out which has size not equal to one
     >>> np.squeeze(x, axis=2).shape
     (1, 3)
+    >>> x = np.array([[1234]])
+    >>> x.shape
+    (1, 1)
+    >>> np.squeeze(x)
+    array(1234)
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py]:
```suggestion
    array(1234). # 0d array
```

File: numpy/core/fromnumeric.py [hunk 3]
```diff
@@ -1438,7 +1438,9 @@ def squeeze(a, axis=None):
 
         Selects a subset of the single-dimensional entries in the
         shape. If an axis is selected with shape entry greater than
-        one, an error is raised.
+        one, an error is raised. Note that when applied to an array
+        `a` with one element and no axis argument, it returns a
+        wrapped scalar of no shape containing the element in `a`.
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py]: Not a huge fan of this phrasing. Perhaps:

 
> Note that if all axes are squeezed, the result is a 0d array and not a scalar.

Comment (eric-wieser) [reply numpy/core/fromnumeric.py]: Also, this comment should move down below to the "Returns" section.

File: numpy/core/fromnumeric.py [hunk 4]
```diff
@@ -1445,7 +1445,8 @@ def squeeze(a, axis=None):
     squeezed : ndarray
         The input array, but with all or a subset of the
         dimensions of length 1 removed. This is always `a` itself
-        or a view into `a`.
+        or a view into `a`.Note that if all axes are squeezed,
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py]:
```suggestion
        or a view into `a`. Note that if all axes are squeezed,
```

**Review by eric-wieser — APPROVED** [review]
Looks good modulo the typo

**Comment by mattip** [comment]
Thanks @Bharat123rox

**Pull request merged**